src/ray/common/client_connection.cc

```diff
@@ -28,8 +28,46 @@
 #include "ray/common/ray_config.h"
 #include "ray/util/util.h"
 
+#if defined(_WIN32)
+#include <Windows.h>
+#else
+#include <fcntl.h>
+#include <unistd.h>
+#endif
+
 namespace ray {
 
+namespace {
+
+#if defined(_WIN32)
+
+// Don't care what exact type is in windows... Looks like to be an asio specific type.
+template <typename NativeHandleType>
+void setFdCloseOnFork(const NativeHandleType &handle) {
+  // In Windows we don't need to do anything, beacuse in CreateProcess we pass
+  // bInheritHandles = false which means we don't inherit handles or sockets.
+  return;
+}
+#else
+void setFdCloseOnFork(int fd) {
+  int flags = fcntl(fd, F_GETFD, 0);
+  if (flags != -1) {
+    fcntl(fd, F_SETFD, flags | FD_CLOEXEC);
```

Comment:
why do you do flags | FD_CLOEXEC instead of just FD_CLOEXEC?

Reply:
this is for furture compatible. An FD may have more flags to set and here we only want to set FD_CLOEXEC. So we read out the other flags and keep them intact. In practice today we only have this FD_CLOEXEC single flag for a FD, but it's common practice and will support the future Linux 7.0 changes.